cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -188,29 +190,71 @@ public ClassLoader getArtifactClassLoader(ArtifactId artifactId) throws IOExcept
     }
   }
 
+  /**
+   * TODO: FIGURE THIS OUT
+   * @param plugin
+   * @param evaluator
+   * @param <T>
+   * @return
+   * @throws IOException
+   * @throws ClassNotFoundException
+   */
+  public <T> T newInstance(Plugin plugin, MacroEvaluator evaluator) throws IOException, ClassNotFoundException,
```

Comment:
to avoid duplicate code, the version that doesn't take a MacroEvaluator should call this method.